Packages/com.unity.inputsystem/InputSystem/Plugins/XR/Daydream.cs

```diff
@@ -0,0 +1,241 @@
+        [FieldOffset(44)]
+        public bool isTracked;
+
+        [InputControl(template = "Vector3")]
+        [FieldOffset(48)]
+        public Vector3 devicePosition;
+
+        [InputControl(template = "Quaternion")]
+        [FieldOffset(60)]
+        public Quaternion deviceRotation;
+
+        [InputControl(template = "Vector3")]
+        [FieldOffset(76)]
+        public Vector3 deviceVelocity;
+
+        [InputControl(template = "Vector3")]
+        [FieldOffset(88)]
+        public Vector3 deviceAcceleration;
+
+        public FourCC GetFormat()
+        {
+            return new FourCC('X', 'R', 'S', '0');
+        }
+    }
+
+    [InputTemplate(stateType = typeof(DaydreamControllerState), commonUsages = new[] { "LeftHand", "RightHand" })]
+    public class DaydreamController : XRController
+    {
+        new public static DaydreamController leftHand { get; private set; }
+        new public static DaydreamController rightHand { get; private set; }
+
+        public Vector2Control touchpad { get; private set; }
+        public ButtonControl volumeUp { get; private set; }
+        public ButtonControl recentered { get; private set; }
+        public ButtonControl volumeDown { get; private set; }
+        public ButtonControl recentering { get; private set; }
+        public ButtonControl app { get; private set; }
+        public ButtonControl home { get; private set; }
+        public ButtonControl touchpadClick { get; private set; }
+        public ButtonControl touchpadTouch { get; private set; }
+
+        public IntegerControl trackingState { get; private set; }
+        public ButtonControl isTracked { get; private set; }
+        public Vector3Control devicePosition { get; private set; }
+        public QuaternionControl deviceRotation { get; private set; }
+        public Vector3Control deviceVelocity { get; private set; }
+        public Vector3Control deviceAcceleration { get; private set; }
+
+        protected override void FinishSetup(InputControlSetup setup)
+        {
+            base.FinishSetup(setup);
+
+            try
+            {
+                XRDeviceDescriptor deviceDescriptor = XRDeviceDescriptor.FromJson(description.capabilities);
+
+                switch (deviceDescriptor.deviceRole)
+                {
+                    case EDeviceRole.LeftHanded:
+                        {
+                            InputSystem.SetUsage(this, CommonUsages.LeftHand);
+                            leftHand = this;
+                            break;
+                        }
+                    case EDeviceRole.RightHanded:
+                        {
+                            InputSystem.SetUsage(this, CommonUsages.RightHand);
+                            rightHand = this;
+                            break;
+                        }
+                    default:
+                        break;
+                }
+            }
+            catch (Exception)
+            { }

```

Comment:
This will need a better solution. I'll take a look at this and SetupWithDescription after merging.

Reply:
Please remove the try/catch block. Exceptions from InputSystem should not be swallowed here.